--- a/squashfs-tools/mksquashfs.c
+++ b/squashfs-tools/mksquashfs.c
@@ -50,9 +50,9 @@
 #include <sys/wait.h>
 #include <limits.h>
 #include <ctype.h>
-#include <sys/sysinfo.h>
 
-#ifdef linux
+#ifdef __linux__
+#include <sys/sysinfo.h>
 #include <sched.h>
 #else
 #include <sys/sysctl.h>
@@ -5081,7 +5081,7 @@
 		BAD_ERROR("Failed to set signal mask in intialise_threads\n");
 
 	if(processors == -1) {
-#ifdef linux
+#ifdef __linux__
 		cpu_set_t cpu_set;
 		CPU_ZERO(&cpu_set);
 
--- a/squashfs-tools/unsquashfs.c
+++ b/squashfs-tools/unsquashfs.c
@@ -2720,7 +2720,7 @@
 	}
 
 	if(processors == -1) {
-#ifdef linux
+#ifdef __linux__
 		cpu_set_t cpu_set;
 		CPU_ZERO(&cpu_set);
 
